connector/connector.go

```diff
@@ -41,6 +44,12 @@ type Traces interface {
 	consumer.Traces
 }
 
+// TracesRouter feeds the first consumer.Traces in each of the specified pipelines.
+type TracesRouter interface {
+	consumer.Traces
+	RouteTraces(context.Context, ptrace.Traces, ...component.ID) error
```

Comment:
Makes sense. Pipeline IDs are component IDs, but with a restriction on DataType. Similar to the relationship between `component.Type` and `component.DataType`.

Reply:
If that is the case what about the Connector factory to accept a "TraceRouter" (probably name can be better):

```golang

type Factory interface {
...

CreateTracesToLogs(ctx context.Context, set CreateSettings, cfg component.Config, next TraceRouter) (Traces, error)

...
}
```

Reply:
Not sure, if this is a better idea, but want to make sure we thought about this.

Reply:
Actually, we can even limit this to "CreateTracesToTraces" to allow fanout routing.

Reply:
> If that is the case what about the Connector factory to accept a "TraceRouter" (probably name can be better):
> 
> ```go
> type Factory interface {
> ...
> 
> CreateTracesToLogs(ctx context.Context, set CreateSettings, cfg component.Config, next TraceRouter) (Traces, error)
> 
> ...
> }
> ```

This is effectively what I tried to accomplish. I allows the signature to remain with `consumer.Traces`, because most connectors will only care about one consumer. But it would be understood to actually be a `connector.TraceRouter` so some connectors can assert the more specific type if they need.

Reply:
> Actually, we can even limit this to "CreateTracesToTraces" to allow fanout routing.

I can think of cross-signal use cases for routing (generating signal but also sending to specific consumers based on configuration). That said, we could limit for now since obvious use cases seem to be same signal.